third-party/realdds/include/realdds/dds-stream.h

```diff
@@ -25,12 +36,25 @@ class dds_stream : public dds_stream_base
 protected:
     dds_stream( std::string const & stream_name, std::string const & sensor_name );
 
-    // dds_stream_base
 public:
-    bool is_open() const override;
-    bool is_streaming() const override;
+    bool is_open() const override { return !! _reader; }
+    virtual void open( std::string const & topic_name, std::shared_ptr< dds_subscriber > const & );
+    virtual void close();

```

Comment:
The device automatically opens the stream, no?
Why would someone need to close it? The topic should remain valid... and closing the streams in the device (meaning stopping streaming) does not mean to disassociate the stream object from the topics...

Reply:
I don't really understand your meaning
The device opens the stream when device::open is called (from `dds_sensor_proxy`).
If the user wants to stop streaming then `close` is called. Destroying the reader better suits your idea of start/stop streaming based on available readers.

The topic can probably be created at construction time (only once), I'll move it

Reply:
Right. The `dds_device` should be the one opening the stream, however it cannot be separated from the reader unless you add another member to the stream, which I don't think you should do.

I missed the point that, in order for the server to start streaming automatically, you cannot have the reader always there since that's what the server picks up on. I think this deserves a comment in the code.